apps/src/templates/teacherDashboard/AssignmentSelector.jsx

```diff
@@ -103,17 +105,23 @@ export default class AssignmentSelector extends Component {
     this.setState({
       selectedPrimaryId,
       selectedSecondaryId
-    });
+    }, this.reportChange);
   };
 
   onChangeSecondary = (event) => {
     this.setState({
       selectedSecondaryId: event.target.value
-    });
+    }, this.reportChange);
+  };
+
+  reportChange = () => {
+    if (typeof this.props.onChange === 'function') {
+      this.props.onChange(this.getSelectedAssignment());
```

Comment:
Yes!  It's not perfectly stateless (it wouldn't immediately pick up changes from an external source) but if all you care about is setting its value on initial render, you can treat it as 'stateless' by passing the `section` (initial value) and `onChange` props.  This lets us tear out all the weird ref-passing that was going on in our earlier PRs.